core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -823,16 +823,14 @@ public void testAddDeleteFilesMultipleSpecs() {
 
     Map<String, String> summary = snapshot.summary();
 
-    Assert.assertEquals("Should change 4 partitions", "4", summary.get(CHANGED_PARTITION_COUNT_PROP));
+    Assert.assertEquals("Should change 3 partitions", "3", summary.get(CHANGED_PARTITION_COUNT_PROP));
     Assert.assertEquals("Should add 1 data file", "1", summary.get(ADDED_FILES_PROP));
     Assert.assertEquals("Should have 4 data files", "4", summary.get(TOTAL_DATA_FILES_PROP));
     Assert.assertEquals("Should add 3 delete files", "3", summary.get(ADDED_DELETE_FILES_PROP));
     Assert.assertEquals("Should have 3 delete files", "3", summary.get(TOTAL_DELETE_FILES_PROP));
     Assert.assertEquals("Should add 3 position deletes", "3", summary.get(ADDED_POS_DELETES_PROP));
     Assert.assertEquals("Should have 3 position deletes", "3", summary.get(TOTAL_POS_DELETES_PROP));
 
-    Assert.assertTrue("Partition metrics must be correct",
-        summary.get(CHANGED_PARTITION_PREFIX).contains(ADDED_DELETE_FILES_PROP + "=1"));
```

Comment:
This partition summary field is for `"partition.": "..."` (the unpartitioned portion of the table).

As we're not writing summaries for unpartitioned fields, this needs to be removed.

We could in theory still add a "global partition" summary if partition summaries are enabled, but update it from `"partition."` to `"partition.unpartitioned"` or something.

Reply:
We could also still increment `changed-partition-count` by 1, given that the `partitions` metadata table _does_ have a row for the "global partition".

Reply:
I think this should still be present. It is okay to have an entry for the table root in cases like this.